community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -545,9 +520,10 @@ public long nextId()
      */
     public void freeId( long id )
     {
-        if ( idGenerator != null )
+        IdGenerator generator = this.idGenerator;
```

Comment:
It's also assigned in `deleteIdGenerator`, which is called during recovery. Plus, it was a memory load. I don't think it's a problem in practice.